### ISSUE TITLE
NavMesh --debug-navigation causing crash on exit outside editor

### ISSUE THREAD

**Issue by lawnjelly** [godotengine/godot#63678]
### Godot version

3.5 RC 8

### System information

Linux Mint 20

### Issue description

Run project directly (not through editor) using the `--debug-navigation` command line, get crash on exit.

Stack trace:
```
1  ArrayMesh::~ArrayMesh                         mesh.cpp                    1455 0x45f6493 
2  memdelete<ArrayMesh>                          memory.h                    114  0x19f00b5 
3  Ref<ArrayMesh>::unref                         reference.h                 258  0x19f0066 
4  Ref<ArrayMesh>::~Ref                          reference.h                 272  0x19ef825 
5  NavigationMesh::~NavigationMesh               navigation_mesh.h           39   0x463e827 
6  memdelete<NavigationMesh>                     memory.h                    114  0x1dda5e5 
7  Ref<NavigationMesh>::unref                    reference.h                 258  0x1dda596 
8  Ref<NavigationMesh>::~Ref                     reference.h                 272  0x1dd0055 
9  NavRegion::~NavRegion                         nav_region.cpp              161  0x1e79fb9 
10 memdelete<NavRegion>                          memory.h                    114  0x1e69f20 
11 GodotNavigationServer::_cmd_free              godot_navigation_server.cpp 589  0x1e6794f 
12 free_command::exec                            godot_navigation_server.cpp 555  0x1e6b8d9 
13 GodotNavigationServer::flush_queries          godot_navigation_server.cpp 620  0x1e634a0 
14 GodotNavigationServer::~GodotNavigationServer godot_navigation_server.cpp 115  0x1e63276 
15 memdelete<NavigationServer>                   memory.h                    114  0x17d6dc5 
16 finalize_navigation_server                    main.cpp                    229  0x17bec80 
17 Main::cleanup                                 main.cpp                    2490 0x17d6a09 
18 main                                          godot_x11.cpp               61   0x177285a 
```
Is crashing because the `VisualServer::get_singleton()` returns NULL by this point.

### Steps to reproduce

Run project with a NavMesh directly (not through editor) using the `--debug-navigation` command line, then quit the app.

**Comment by lawnjelly** [comment]
Marked as 3.x because it shouldn't be a showstopper for 3.5 imo. It's a kind of rare situation, and easy to fix in a point release.

**Comment by akien-mga** [comment]
Fixed by #70343.
